rs/embedders/src/wasm_utils/instrumentation.rs

```diff
@@ -1723,17 +1727,34 @@ fn get_data(
                         "complex initialization expressions for data segments are not supported!".into()
```

Comment:
Is this just an existing wasm restriction or an additional one we are imposing. I hope the former.

Reply:
This function only extracts the data of active data segments such that the IC loads them in the initial main memory state. Passive data segments can and should be ignored here because they will be loaded explicitly by the Wasm code.
The `unreachable` looks worse than it is, the passive data segments are excluded by the preceding `filter()`. So, this condition should never be met (only included because of the Rust compiler).